controllers/volumereplication_controller.go

```diff
@@ -52,7 +56,72 @@ type VolumeReplicationReconciler struct {
 func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = r.Log.WithValues("volumereplication", req.NamespacedName)
 
-	// your logic here
+	reconcileResponse, err := r.reconcile(req)
+
+	if err != nil {
+		r.Log.Error(err, "Failed to reconcile")
+	}
+
+	return reconcileResponse, err
+}
+
+func (r *VolumeReplicationReconciler) reconcile(req ctrl.Request) (ctrl.Result, error) {
+
+	// Get volume replication
+	vrObj := &replicationv1alpha1.VolumeReplication{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, vrObj)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("VolumeReplication not found", "object", req.NamespacedName)
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// Get volume replication class
+	vrcObj := &replicationv1alpha1.VolumeReplicationClass{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: vrObj.Spec.VolumeReplicationClass}, vrcObj)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Error(err, "VolumeReplicationClass not found", "object", vrObj.Spec.VolumeReplicationClass)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, err
+	}
+
+	if r.DriverConfig.DriverName != vrcObj.Spec.Provisioner {
+		err = fmt.Errorf("VolumeReplicationClass Provisioner %s does not match with %s", vrcObj.Spec.Provisioner, r.DriverConfig.DriverName)
+		r.Log.Error(err, "Provisioner does not match")
+		return ctrl.Result{}, err
```

Comment:
As the driver name is not matching we should not log or return an error we just need to return success here